src/Transport.ts

```diff
@@ -331,320 +338,383 @@ export default class Transport {
+        } else {
+          headers['content-length'] = '' + Buffer.byteLength(connectionParams.body) // eslint-disable-line
         }
-        headers['content-encoding'] = 'gzip'
-        headers['content-length'] = '' + Buffer.byteLength(connectionParams.body) // eslint-disable-line
-      } else {
-        headers['content-length'] = '' + Buffer.byteLength(connectionParams.body) // eslint-disable-line
       }
-    }
 
-    headers.accept = headers.accept ?? this[kAcceptHeader]
-    connectionParams.headers = headers
-    while (meta.attempts <= maxRetries) {
-      try {
-        if (signal?.aborted) { // eslint-disable-line
-          throw new RequestAbortedError('Request has been aborted by the user', result, errorOptions)
-        }
+      headers.accept = headers.accept ?? this[kAcceptHeader]
+      connectionParams.headers = headers
+      while (meta.attempts <= maxRetries) {
+        try {
+          if (signal?.aborted) { // eslint-disable-line
+            throw new RequestAbortedError('Request has been aborted by the user', result, errorOptions)
+          }
 
-        meta.connection = this.getConnection({
-          requestId: meta.request.id,
-          context: meta.context
-        })
-        if (meta.connection === null) {
-          throw new NoLivingConnectionsError('There are no living connections', result, errorOptions)
-        }
+          meta.connection = this.getConnection({
+            requestId: meta.request.id,
+            context: meta.context
+          })
+          if (meta.connection === null) {
+            throw new NoLivingConnectionsError('There are no living connections', result, errorOptions)
+          }
 
-        this[kDiagnostic].emit('request', null, result)
-
-        // perform the actual http request
-        let { statusCode, headers, body } = await meta.connection.request(connectionParams, {
-          requestId: meta.request.id,
-          name: this[kName],
-          context: meta.context,
-          maxResponseSize,
-          maxCompressedResponseSize,
-          signal,
-          timeout: toMs(options.requestTimeout != null ? options.requestTimeout : this[kRequestTimeout]),
-          ...(options.asStream === true ? { asStream: true } : null)
-        })
-        result.statusCode = statusCode
-        result.headers = headers
-
-        if (this[kProductCheck] != null && headers['x-elastic-product'] !== this[kProductCheck] && statusCode >= 200 && statusCode < 300) {
-          /* eslint-disable @typescript-eslint/prefer-ts-expect-error */
-          // @ts-ignore
-          throw new ProductNotSupportedError(this[kProductCheck], result, errorOptions)
-          /* eslint-enable @typescript-eslint/prefer-ts-expect-error */
-        }
+          // generate required OpenTelemetry attributes from the request URL
+          const requestUrl = meta.connection.url
+          otelSpan?.setAttributes({
+            'url.full': requestUrl.toString(),
```

Comment:
This may be a spec issue. It describes the `url.full` attribute as being "Required". However, in some error cases (e.g. `throw new NoLivingConnectionsError` above), it is possible it isn't set on a span.  I don't think there is anything you need do about this in this PR.